Create hoda.alkhatib (#143)
UPB-FILS-MA/upb-fils-ma.github.io: docs/project/2024/hoda.alkhatib/index.md

Maintains current game state and information.
Facilitates communication between components.
User Interface (UI):
Provides visual indicators and feedback.
Enhances player engagement and experience.

-   how they connect with each other

The Raspberry Pi is positioned at the center, serving as the main controller.
Components are arranged in a clear and organized manner around the Raspberry Pi.
Arrows indicate the direction of data flow or connection from the components to the Raspberry Pi.
Each component is labeled for clarity.

## Log

<!-- write every week your progress here -->

### Week 6 - 12 May

### Week 7 - 19 May

### Week 20 - 26 May

## Hardware

My hardware setup comprises a Raspberry Pi Pico microcontroller, a breadboard serving as the main board, an 8×8 LED matrix module, a passive buzzer, a joystick module, and various wires for connection, a LED , a resistor.

### Schematics

The Raspberry Pi is positioned at the center, serving as the main controller.
Components are arranged in a clear and organized manner around the Raspberry Pi.
Arrows indicate the direction of data flow or connection from the components to the Raspberry Pi.
Each component is labeled for clarity.

> ![image](https://github.com/AlkhatibHoda/upb-fils-ma.github.io/assets/163418706/6d8fd589-afbd-4f42-a0ab-eaa5cbad18c2)

Reviewer: ```suggestion
```

# Snake game project 

> 

Reviewer: Please add a one line description of the project.

:::info

> Author: Alkhatib Hoda
> 
> GitHub Project Link: https://github.com/UPB-FILS-MA/project-AlkhatibHoda

Reviewer: ```suggestion
**Author**: Alkhatib Hoda \
**GitHub Project Link**: https://github.com/UPB-FILS-MA/project-AlkhatibHoda
```

:::

## Description

My project is a recreation of the popular game 'Snake,' displayed on an 8×8 LED Matrix module using a Raspberry Pi.My aim is to create something that appears simple but holds many memories of my childhood and provides enjoyable entertainment.

## Motivation

Creating a Snake game with Rust on a Raspberry Pi and LED matrix blends nostalgia with technical challenge, fostering both creativity and learning.

## Architecture

Add here the schematics with the architecture of your project. Make sure to include:

-   what are the main components (architecture components, not hardware components)
the main architecture components:

> Game Engine:

Reviewer: Does this render correctly? Did you try using `npm start`?

Maintains current game state and information.
Facilitates communication between components.
User Interface (UI):
Provides visual indicators and feedback.
Enhances player engagement and experience.

-   how they connect with each other

The Raspberry Pi is positioned at the center, serving as the main controller.
Components are arranged in a clear and organized manner around the Raspberry Pi.
Arrows indicate the direction of data flow or connection from the components to the Raspberry Pi.
Each component is labeled for clarity.

## Log

<!-- write every week your progress here -->

### Week 6 - 12 May

### Week 7 - 19 May

### Week 20 - 26 May

## Hardware

My hardware setup comprises a Raspberry Pi Pico microcontroller, a breadboard serving as the main board, an 8×8 LED matrix module, a passive buzzer, a joystick module, and various wires for connection, a LED , a resistor.

### Schematics

The Raspberry Pi is positioned at the center, serving as the main controller.
Components are arranged in a clear and organized manner around the Raspberry Pi.
Arrows indicate the direction of data flow or connection from the components to the Raspberry Pi.
Each component is labeled for clarity.

> ![image](https://github.com/AlkhatibHoda/upb-fils-ma.github.io/assets/163418706/6d8fd589-afbd-4f42-a0ab-eaa5cbad18c2)

Reviewer: Please add the image to the folder and use it.

Place your KiCAD schematics here.

### Bill of Materials

<!-- Fill out this table with all the hardware components that you might need.

The format is

| [Device](link://to/device) | This is used ... | [price](link://to/store) |

-->

| Device                                                                                                  | Usage               | Price                                                                                                                                                                                                                                                                                |

> | ------------------------------------------------------------------------------------------------------- | ------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

Reviewer: ```suggestion
|-|-|-|
```

| [Passive Buzzer]()                                                                                      | Buzzer              | [3,97 RON ](https://ardushop.ro/ro/electronica/194-buzzer.html?search_query=buzzer&results=16)                                                                                                                         |
| [Joystick mode]()                                                                                       | control the game    | [3,60 RON ](https://ardushop.ro/ro/electronica/127-modul-joystick.html?search_query=joystick&results=4)                                                                                                                                                                                                                                                         |

> | [A LED]()                                                                                               | turns on when starting | [0 RON( I borrowed it)]()                                                                                                                                                                                                                                                          |

Reviewer: This link is not valid.

| [40 x Dupont Yarn Mother-Father 10cm]()                                                                 | connection           | [5,18  RON](https://ardushop.ro/ro/electronica/23-40-x-dupont-cables-female-male-10cm.html?search_query=fir&results=286) |

## Software

| Library                                  | Description                        | Usage                              |
| ---------------------------------------- | ---------------------------------- | ---------------------------------- |
| [VsCode](https://code.visualstudio.com/) | Integrated Development Environment | Integrated Development Environment |
| [Rust](https://www.rust-lang.org/)       | Programming Language               | Programming Language               |
| [Embassy](https://embassy.dev/)          | Framework                          | Framework                          |

More to come as we are developing our project!

## Links

<!-- Add a few links that inspired you and that you think you will use for your project -->

> 1. [from last year projects](https://ocw.cs.pub.ro/courses/pm/prj2023/apredescu/gameofsnake)

Reviewer: Please use the project's title for the link title.

# Snake game project 

> The game 'Snake,' displayed on an 8×8 LED Matrix module 

Reviewer: ```suggestion
The game 'Snake', displayed on an 8×8 LED Matrix module 
```
